webrender_api/src/display_list.rs

```diff
@@ -236,35 +236,46 @@ impl<'a> BuiltDisplayListIter<'a> {
         self.cur_clip_chain_items = ItemRange::default();
 
         loop {
-            if self.data.is_empty() {
-                return None;
+            self.next_raw();
+            if let SetGradientStops = self.cur_item.item {
+                // SetGradientStops is a dummy item that most consumers should ignore
```

Comment:
can we leave this decision to the clients, given that they do know which items they need and which they don't?

Reply:
They can just use `_raw` if they want to make that decision, right? This method is for the "intended" case, where SetGradientStops is really just a quirk of our encoding.